tests/unit/download/mocks.ts

```diff
@@ -0,0 +1,27 @@
+import { vi } from "vitest";
+
+export const mockWorkbook = { sheets: [] } as any;
+export const mockAoaToSheet = vi.fn();
+export const mockBookNew = vi.fn();
+export const mockWriteFile = vi.fn();
+export const mockBookAppendSheet = vi.fn();
+export const mockJsonToSheet = vi.fn();
+
+vi.mock("xlsx", () => ({
+  writeFile: (data: string, fileName: string) => mockWriteFile(data, fileName),
+  utils: {
+    aoa_to_sheet: (data: any) => mockAoaToSheet(data),
+    json_to_sheet: (data: any) => mockJsonToSheet(data),
+    book_new: () => mockBookNew(),
+    book_append_sheet: (wb: any, ws: any, name: string) => mockBookAppendSheet(wb, ws, name),
+  },
+}));
+
+export const mockImplementations = () => {
+  mockJsonToSheet.mockImplementation(data => ({ data, type: "json" }));
```

Comment:
I guess you are replacing the implementations of the private methods because there is no way to create and assert on a real xlsx file during a test?

Reply:
Yeah, I think we can trust xlsx package to do what we expect. Could have included an integration test that wrote out, then read in and checked a real xlsx file, but that kind of seemed like overkill...